onnxruntime/contrib_ops/cuda/bert/packed_multihead_attention.cc

```diff
@@ -250,15 +272,15 @@ Status PackedMultiHeadAttention<T>::ComputeInternal(OpKernelContext* context) co
   constexpr size_t element_size = sizeof(T);
   // When the source and target format is same (like TN3H => TN3H, or TNH => TNH) and no bias, need not transpose qkv.
   const bool no_qkv_workspace = (fused_runner != nullptr && key == nullptr && bias == nullptr) ||
-                                (use_memory_efficient_attention && value != nullptr && bias == nullptr);
+                                ((use_memory_efficient_attention || use_flash_attention) && value != nullptr && bias == nullptr);
   size_t workSpaceSize = GetAttentionWorkspaceSize(element_size,
                                                    parameters.batch_size,
                                                    parameters.num_heads,
                                                    parameters.head_size,
                                                    parameters.v_head_size,
                                                    parameters.sequence_length,
                                                    fused_runner,
-                                                   use_memory_efficient_attention,
+                                                   use_memory_efficient_attention || use_flash_attention,
```

Comment:
Please pass use_flash_attention as a separated parameter to GetAttentionWorkspaceSize.